### PR TITLE
Remove BaseProcessorFactory, force using helper

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- Remove BaseProcessorFactory and add internalinterface.InternalInterface to force people to use the helper.

**Link to tracking Issue:** Updates #3921

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@Aneurysm9 you had to think about this. Do you think this is the best approach?

**Comment by mx-psi** [comment]
@bogdandrutu done!

File: component/processor_test.go [hunk 1]
```diff
@@ -40,6 +43,21 @@ func (f *TestProcessorFactory) CreateDefaultConfig() config.Processor {
...
+
+// CreateMetricsProcessor default implemented as not supported data type.
+func (f *TestProcessorFactory) CreateMetricsProcessor(context.Context, ProcessorCreateSettings, config.Processor, consumer.Metrics) (MetricsProcessor, error) {
+	return nil, componenterror.ErrDataTypeIsNotSupported
+}
+
+// CreateLogsProcessor default implemented as not supported data type.
+func (f *TestProcessorFactory) CreateLogsProcessor(context.Context, ProcessorCreateSettings, config.Processor, consumer.Logs) (LogsProcessor, error) {
+	return nil, componenterror.ErrDataTypeIsNotSupported
+}
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply component/processor_test.go:59]:
```
type TestProcessorFactory struct {
	ProcessorFactory
	name string
}
```

Then no need to declare all the Create funcs.

[the remark this answers is not in the document]
Comment (mx-psi) [reply component/processor_test.go:59]: 👍, see #4336

**Pull request merged**